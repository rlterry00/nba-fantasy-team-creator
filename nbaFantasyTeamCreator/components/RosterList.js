import React from 'react';
import {StyleSheet, Text, View, Button, Image} from 'react-native';

const RosterList = props => {
  return (
    <View style={styles.listContainer}>
      <View>
        <View style={styles.titleViewRowRoster}>
          <Text style={styles.sectionTitle}>{props.cityName}</Text>
          <Text style={styles.sectionTitle}>{props.teamName}</Text>
        </View>
        <Text style={styles.sectionTitle}>
          Position: {props.playerPosition}
        </Text>
        <View style={styles.nameRow}>
          <Image
            style={{width: 100, height: 100, marginLeft: 5}}
            defaultSource={require('../assets/avatar.png')}
            source={{uri: props.playerImage}}
            resizeMode="cover"
          />

          <Text style={styles.sectionTitle}>{props.firstName}</Text>
          <Text style={styles.sectionTitle}>{props.lastName}</Text>
        </View>

        <View style={styles.button}>
          <Button
            onPress={() => props.onPress()}
            title="Remove Player"
            color="black"
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  listContainer: {
    flex: 1,
    margin: 10,
  },
  titleViewRowRoster: {
    flexDirection: 'row',
  },
  title: {
    fontSize: 30,
  },
  sectionTitle: {
    fontSize: 25,
    marginRight: 5,
    marginLeft: 5,
    marginTop: 10,
  },
  nameRow: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    borderBottomWidth: 1,
    marginBottom: 5,
  },
  button: {
    backgroundColor: 'gainsboro',
    borderRadius: 15,
    margin: 10,
  },
});
export default RosterList;
