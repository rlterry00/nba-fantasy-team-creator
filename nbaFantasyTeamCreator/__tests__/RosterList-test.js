import 'react-native';
import React from 'react';
import RosterList from '../components/RosterList';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<RosterList />);
});

test('matches snapshot', () => {
  const rosterList = renderer.create(<RosterList />).toJSON();
  expect(rosterList).toMatchSnapshot();
});
