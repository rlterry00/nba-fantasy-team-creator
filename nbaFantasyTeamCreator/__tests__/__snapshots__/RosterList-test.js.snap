// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 25,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 25,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 10,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "fontSize": 25,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 10,
        }
      }
    >
      Position: 
    </Text>
    <View
      style={
        Object {
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginBottom": 5,
        }
      }
    >
      <Image
        defaultSource={
          Object {
            "testUri": "../../../assets/avatar.png",
          }
        }
        resizeMode="cover"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": 100,
            "marginLeft": 5,
            "width": 100,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 25,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 25,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "gainsboro",
          "borderRadius": 15,
          "margin": 10,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Remove Player
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
